pkg/webhook/conversion/decoder.go

```diff
@@ -0,0 +1,30 @@
+package conversion
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+// Decoder knows how to decode the contents of a CRD version conversion
+// request into a concrete object.
+type Decoder struct {
+	codecs serializer.CodecFactory
+}
+
+// NewDecoder creates a Decoder given the runtime.Scheme
+func NewDecoder(scheme *runtime.Scheme) (*Decoder, error) {
+	return &Decoder{codecs: serializer.NewCodecFactory(scheme)}, nil
+}
+
+// Decode decodes the inlined object.
+func (d *Decoder) Decode(content []byte) (runtime.Object, *schema.GroupVersionKind, error) {
```

Comment:
I feel we can move [the decoder in admission pkg](https://github.com/kubernetes-sigs/controller-runtime/blob/8d94f663b1f552f74805cd8c44a1e03387f5a5d2/pkg/webhook/admission/decode.go#L44) to a shared the place and reuse it potentially.

Reply:
Added a TODO for now. I am sure `conversion` pkg will evolve as it goes to `Beta` and we can reevaluate it at that point.

Reply:
not convinced we need this at all (see above comment)